### ISSUE TITLE
Mantid EventWorkspace to bucket conversion

### ISSUE THREAD

**Issue by SimonHeybrock** [scipp/scipp#1335]
Currently Mantid's `EventWorkspace` gets converted to variables with, e.g., `dtype=event_list_double`. With #1281 event lists will be removed and replaced by bucket variables. While support for this is not complete yet we can already implement the converter (while keeping the old one in parallel).

- Concatenate all event lists into a single data array.
- Use `to_buckets` to create a bucket variable from this.
  - We may initially add just a single large bucket along the "tof" dimension, i.e., `dims=['spectrum', 'tof']` and `shape=[nspec, 1]`. Ultimately we may want to split buckets along `tof`. This requires sorting with buckets and finding the indices. There are no tools for this implemented yet, so the suggestion would be to postpone this.
- The bucket variable will be the "data" of the data array created from the workspace. Remaining coords and masks are added as usual.
  - The bin edges for `tof` would only use the first and last edge, as long as we support only a single bucket along `tof`.
- For now, the simplest solution may be to add a flag to the converter, to chose how event data gets converted. Later, we remove the old conversion and the flag.

**Comment by OwenArnold** [comment]
@DanNixon are you actively working on this? If not it would be useful to assign to someone else.

**Comment by DanNixon** [comment]
I was before I went on leave. It will be at least next week before it is done so if it needs done before then go ahead and reassign it.
